crates/utils/src/tracing/span_ext.rs

```diff
@@ -48,21 +59,53 @@ impl ToValue for i64 {
 ///
 /// This is a sealed trait. It and cannot be implemented outside of this module.
 pub trait OpenTelemetrySpanExt: private::Sealed {
+    fn set_parent<T>(&self, request: &http::Request<T>);
     fn set_attribute(&self, key: impl Into<Key>, value: impl ToValue);
+    fn set_http_attributes<T>(&self, request: &http::Request<T>);
     fn set_error(&self, err: &dyn std::error::Error);
+    fn context(&self) -> opentelemetry::Context;
```

Comment:
I wouldn't add these to the trait. Bear in mind that this trait is used by all other crates - and they have no use for setting the parent from a request, nor http attributes nor context. In fact doing so would be an error / mistake as its only useful for tracing gRPC entry-points/requests.

In other words they should remain private / internal, which probably means they should just revert to free standing functions.